code-challenges.test.js

```diff
@@ -13,47 +16,151 @@
+//now after my changes it returns that array.map.toUpperCase() isn't a function. not sure what i might be over looking or over thinking it.
+const randomNouns2 = ["temperature", "database", "chopsticks", "mango"];
+// Expected output: ["Temperature", "Database", "Chopsticks", "Mango"]
+//i will write the test for this even though my jest doesnt seem to be aggreeing with me.
+describe("capArray", () => {
+  it("takes an array with strings and returns an array with the strings capitalized", () => {
+    expect(capArray()).toEqual(array.toUpperCase());
+  });
+});
 // b) Create the function that makes the test pass.
-
+// reusing the function i created. i believe i have hit the point in my testing that is running into syntax and operating problems due to the information in this file. i will restart terminal and comment out all information that doesn't obtain to this specific test..
+// const capArray = (array) => {
+//   return array().map("string").toUpperCase();
+// };
+// test result failed to run. it says the suite must contain at least one test.. moving forward
 // --------------------3) Create a function that takes in a string and logs the index of the first vowel.
-
+//input= string
+//expected output= an index of the first vowel
+//i think a for loop that iterates thru the string and returns the first vowel of the string
 // a) Create a test with expect statements for each of the variables provided.
-
-const vowelTester1 = "learn"
-// Expected output: 1
-const vowelTester2 = "academy"
-// Expected output: 0
-const vowelTester3 = "challenges"
-// Expected output: 2
-
-// b) Create the function that makes the test pass.
+//first i created the test describing the function it will test and what the expected output template
+describe("firstVowel", () => {
+  it("returns the index of the first vowel of the string", () => {
+    expect(firstVowel(vowelTester1)).toEqual(1);
+    expect(firstVowel(vowelTester2)).toEqual(0);
+    expect(firstVowel(vowelTester3)).toEqual(2);
+  });
+});
+
+const vowelTester1 = "learn";
+// // Expected output: 1
+const vowelTester2 = "academy";
+// // Expected output: 0
+const vowelTester3 = "challenges";
+// // Expected output: 2
+//going with now way to test and see if my answers are right or not, i am hoping that i coded this right. i will figure out(in the near future) what is blocking me from completing this assesment properly and make sure this doesn't happen again.
+
+// // b) Create the function that makes the test pass.
+function firstV(str) {
+  let test = str.search(/[a,e,i,o,u,A,E,I,O,U]/g);
+}
+
+// I know i want to search for the vowel and show the index of the first vowel. yet with my research, this is the function i came up with. i look forward to feed back and to figure out the better way to complete this weeks assesment. thank you for your time.
```

Comment:
This is SUPER close!  Don't forget every function needs a return.  When using new concepts that we have not covered, make sure to include definitions in your pseudocode on what they are and why you chose to use it.  For instance, inside the search, you are using a great tool called Regex.